### ISSUE TITLE
ConstrainLong only concern positive values

### ISSUE THREAD

**Issue by dontcallmedom** [w3c/mediacapture-main#385]
As far as I can tell, all the usage of the `ConstrainLong` types refers to values that can only be positive (`width`, `height`, `sampleSize`, `sampleRate`, `channelCount`).

Should we be using `unsigned long` rather than `long` then? (and probably rename it to `ConstrainUnsignedLong`

**Comment by alvestrand** [comment]
Unless we use [Clamp] or [EnforceRange], this will turn the error of passing -50 to a constraint into a barely-noticeable +50, if I read http://es5.github.io/#x9.5 correctly.

I see this as a Bad Thing.

**Comment by dontcallmedom** [comment]
No, if you pass -50, it will throw a TypeError per [WebIDL conversion algorithm of `unsigned long`](https://www.w3.org/TR/WebIDL-1/#es-unsigned-long)

**Comment by alvestrand** [comment]
I wish it did - and I was talking about the non-attribute case. When it's not annotated, Steps 2 and 3 are skipped because there is neither [Clamp] nor [EnforceRange], and passes to ECMAScript's ToUInt32 algorithm, which happily matches -50 to 50.

**Comment by dontcallmedom** [comment]
woops, indeed, you're right; so clearly we should only do this if #384 gets resolved positively :)
